lib/LMSManagers/LMSHelpdeskManager.php

```diff
@@ -1258,71 +1258,105 @@ public function GetTicketContents($id, $short = false)
+        if (empty($ticket['requestor_userid'])
+            && (!empty($ticket['requestor'])
+                || !empty($ticket['requestor_mail'])
+                || !empty($ticket['requestor_phone']))) {
             $ticket['requestor_userid'] = 0;
         }
 
-        $ticket['categories'] = $this->db->GetAllByKey('SELECT categoryid AS id, c.name, c.style
-								FROM rtticketcategories tc
-								JOIN rtcategories c ON c.id = tc.categoryid
-								WHERE ticketid = ?', 'id', array($id));
-        $ticket['categorynames'] = empty($ticket['categories']) ? array() : array_map(function ($elem) {
+        $ticket['categories'] = $this->db->GetAllByKey(
+            'SELECT categoryid AS id, c.name, c.style
+                FROM rtticketcategories tc
+                JOIN rtcategories c ON c.id = tc.categoryid
+                WHERE ticketid = ?',
+            'id',
+            array($id)
+        );
+
+        $ticket['categorynames'] = empty($ticket['categories']) ? array() :
+            array_map(function ($elem) {
                 return $elem['name'];
-        }, $ticket['categories']);
+            }, $ticket['categories']);
 
         $ticket['parent'] = $this->getTickets($ticket['parentid']);
         $ticket['relatedtickets'] = $this->GetRelatedTickets($id);
 
         if (!$short) {
-            $ticket['messages'] = $this->db->GetAll(
-                '(SELECT rtmessages.id AS id, phonefrom, mailfrom, subject, body, contenttype, createtime, '
-                . $this->db->Concat('customers.lastname', "' '", 'customers.name') . ' AS customername,
-					userid, vusers.name AS username, customerid, rtmessages.type, rtmessages.deleted, rtmessages.deltime, rtmessages.deluserid
-					FROM rtmessages
-					LEFT JOIN customers ON (customers.id = customerid)
-					LEFT JOIN vusers ON (vusers.id = userid)
-					WHERE 1=1'
-                . (!ConfigHelper::checkPrivilege('helpdesk_advanced_operations') ? ' AND rtmessages.deleted = 0' : '')
-                . (' AND ticketid = ?)')
-                . (' ORDER BY createtime ASC, rtmessages.id'),
-                array($id)
-            );
+            $ticket['messages'] = $this->GetTicketMessages(array('id' => $id));
+        }
+
+        $ticket['status'] = $RT_STATES[$ticket['state']];
+        $ticket['uptime'] = uptimef($ticket['resolvetime'] ?
+            $ticket['resolvetime'] - $ticket['createtime'] :
+            time() - $ticket['createtime']);
+
+        if (!empty($ticket['nodeid']) && empty($ticket['node_location'])) {
+            $customer_manager = new LMSCustomerManager($this->db, $this->auth, $this->cache, $this->syslog);
+            $ticket['node_location'] = $customer_manager->getAddressForCustomerStuff($ticket['customerid']);
+        }
+        return $ticket;
+    }
+
+    public function GetTicketMessages($params)
```

Comment:
Tę metodę na zewnątrz klasy z założenia będzie do użycia?

Reply:
Na tę chwilę tego nie potrzebujemy nigdzie - ale w przyszłości tak.

Reply:
> Tę metodę na zewnątrz klasy z założenia będzie do użycia?

Wyjaśnisz dlaczego o to pytasz?
Mogę zmienić widoczność na `private` bo metoda nie jest nigdzie indziej używana.